### PR TITLE
[SG-69] Billing payment step

### PR DESCRIPTION
**Pull request by gbubemismith**
## Type of change

```
- [ ] Bug fix
- [X] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective
Create the billing and payment step for the new trial initiation feature.

## Code changes

- **apps/web/src/app/modules/billing/billing.component.html:** Billing template component which reuses  `app-payment` and `app-tax-info` component.
- **apps/web/src/app/modules/billing/billing.component.ts:** Uses  `OrganizationPlansComponent` as the base class since it has most of the logic to successfully make a payment. The class takes `orgInfoForm`, `plan`, `product` as inputs, and `previousStep` output which is used to go back to the previous step.
- **apps/web/src/app/modules/trial-initiation/trial-initiation.component.html:** Added the billing component step to the trail initiation.
- **apps/web/src/app/modules/trial-initiation/trial-initiation.component.ts:** query parameter from the URL is used to determine the plan and product which is then passed as an input to the billing component.
- **apps/web/src/app/settings/organization-plans.component.ts:** Emit event with `orgId` and `billingSubLabelText` on successful payment. Created `billingSubLabelText` method to return the `billingSubLabelText` .
- **apps/web/src/app/settings/payment.component.html:** Added `trialFlow` input which is used with `[ngClass]` to alter the grid system for trial initiation.
- **apps/web/src/app/settings/payment.component.ts:** Added `trialFlow` input which is used with `[ngClass]` to alter the grid system for trial initiation.
- **apps/web/src/locales/en/messages.json:** Added locales needed for the billing and payment step.
- **libs/angular/src/components/register.component.ts:** Added `login` call after successfully creating an account. This background login is necessary to call most endpoints in the other steps that need authorization.

Screenshot 2022-07-19 at 2 26 55 PM
Screenshot 2022-07-19 at 2 27 22 PM

## Before you submit

```
- [X] I have checked for **linting** errors (`npm run lint`) (required)
- [ ] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)
```

### CODE REVIEW

File: apps/web/src/app/settings/tax-info.component.html [hunk 1]
```diff
@@ -265,7 +265,7 @@
       </select>
     </div>
   </div>
-  <div class="col-3">
+  <div class="col-4">
```
Comment (differsthecat) [inline apps/web/src/app/settings/tax-info.component.html:268]: Does this change look ok on the existing component?

File: apps/web/src/app/modules/billing/billing.component.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      additionalSeats: 1,
+      plan: this.plan,
+      product: this.product,
+    });
+    this.isInTrialFlow = true;
+    await super.ngOnInit();
+  }
+
+  stepBack() {
+    this.previousStep.emit(null);
```
Comment (differsthecat) [inline apps/web/src/app/modules/billing/billing.component.ts:66]: I'm pretty sure you don't have to pass null if the event emitter doesn't have a value
```suggestion
    this.previousStep.emit();
```

File: apps/web/src/app/modules/trial-initiation/trial-initiation.component.ts [hunk 3]
```diff
@@ -40,6 +47,17 @@ export class TrialInitiationComponent implements OnInit {
...
+      if (qParams.org === "families") {
+        this.plan = PlanType.FamiliesAnnually;
+        this.product = ProductType.Families;
+      } else if (qParams.org === "teams") {
+        this.plan = PlanType.TeamsAnnually;
+        this.product = ProductType.Teams;
+      } else if (qParams.org === "enterprise") {
+        this.plan = PlanType.EnterpriseAnnually;
+        this.product = ProductType.Enterprise;
+      }
```
Comment (differsthecat) [inline apps/web/src/app/modules/trial-initiation/trial-initiation.component.ts:60]: Is there any issue with putting the org checks into the above if check on line 47? We don't need to run through these checks if the parameter is not defined.

**Review by differsthecat — COMMENTED** [review]
A few comments; Can you also add some screenshots to this, with both light and dark mode?

File: apps/web/src/app/settings/tax-info.component.html (hunk 1 above)
Comment (gbubemismith) [reply apps/web/src/app/settings/tax-info.component.html:268]: Yes, it does. I would change this to use `[ngClass]`.

File: apps/web/src/app/modules/trial-initiation/trial-initiation.component.html [hunk 4]
```diff
@@ -58,10 +58,14 @@ <h2 class="tw-uppercase tw-pl-4 tw-text-base tw-mb-0 tw-font-bold">
               </button>
             </app-vertical-step>
             <app-vertical-step label="Billing">
```
Comment (differsthecat) [inline apps/web/src/app/modules/trial-initiation/trial-initiation.component.html:60]: It doesn't look like the billing sub label is passed here from the screenshots, but you have the logic for it in the typescript

Comment (gbubemismith) [reply apps/web/src/app/modules/trial-initiation/trial-initiation.component.html]: thanks, that was mistakenly ommited

**Review by differsthecat — APPROVED** [review]
Looks good, excellent work @gbubemismith !

**Pull request merged**